### PR TITLE
Mage_Usa: added "USPS Ground Advantage" and removed "First Class Package"

### PR DESCRIPTION
**Pull request by fballiano**
Untested, taken from https://github.com/OpenMage/magento-lts/discussions/3300#discussioncomment-6567705

### CODE REVIEW

**Comment by sreichel** [comment]
Untested? Just to have another commit?

**Comment by Flyingmana** [comment]
> Untested? Just to have another commit?

This response has no substance and contains apparently a personal attack, see this as a Warning for a Code of Conduct Violation

**Comment by fballiano** [comment]
It's only a way to take a user's discussion (you can read that there were interactions in order to understand what we should do) and put it into code in order not to let it be forgotten.
I've always tested (as far as my abilities go) other's people code and mine, in this case I can't test because I don't have enough knowledge about USPS systems to understand if it's working as intended.

**Comment by sreichel** [comment]
> > Untested? Just to have another commit?
> 
> This response has no substance and contains apparently a personal attack, see this as a Warning for a Code of Conduct Violation

Thats just wrong.

Why to commit untested code? Cant understand ... QA?

**Comment by ADDISON74** [comment]
We focus the discussion on the word "**untested**" instead of losing sight of the big picture. When I came here, I kept hearing very often the phrase "if you have an idea, create a PR" from the maintenance side. I believe that any proposed PR starts from the beginning with some predetermined conditions such as: it is not a problem in the code, yes it is a problem but it does not solve it. Here, everyone can be the devil's advocate and do well as long as he has solid arguments.

Fabrizio took a code suggested by someone and mentioned that it is untested. In order for this PR to be approved, it needs at least 2 green, 1 green and 2 gray approvals. As you can see, no one has approved it, if they do, it means that they have tested it and can provide feedback. The fact that he opened it is beneficial, those with similar problems can checkout and determine if something is solved or not.

Let's not get attached to one word in a sentence. I pretend that we are mature people.

**Comment by fballiano** [comment]
OMFG I CAN'T HANDLE THIS **** NO MORE!!!!!!!!!!!!!

**Comment by fballiano** [comment]
tested with a customer which confirms this patch solves the problem!

Screenshot 2023-08-04 alle 08 45 13

this should be released as soon as possible because at the moment whoever's using USPS is missing on the cheaper shipping options and it is a big deal.

**Comment by fballiano** [comment]
@michaelj000 @johnvoncolln

**Comment by johnvoncolln** [comment]
@fballiano this is great, thanks for doing it!

**Comment by fballiano** [comment]
it seems we have 2 positive tests, can we please try to merge this?

File: app/code/core/Mage/Usa/Model/Shipping/Carrier/Usps.php
Comment (fballiano) [reply app/code/core/Mage/Usa/Model/Shipping/Carrier/Usps.php:529]: mmmm magento/magento2#37745 says that first class is replaced by ground advantage too. i've no way of verifying, the only feedback that I have is from a a customer and a user who write here and they're positive.

Comment (kiatng) [reply app/code/core/Mage/Usa/Model/Shipping/Carrier/Usps.php:529]: You are right, the grace period of products

* First-Class Package Service
• First-Class Package Return Service
• Parcel Select Ground
• Ground Returns

will expire on September 30, 2023.

Comment (fballiano) [reply app/code/core/Mage/Usa/Model/Shipping/Carrier/Usps.php:529]: I think this should be merged in order to be released in a couple of months realistically. who uses USPS (like a new customer of mine) already had to apply this patch or the USPS module was broken (since the ground shipping is usually the cheapest)

**Comment by fballiano** [comment]
@johnvoncolln could you please approve this PR at this link: https://github.com/OpenMage/magento-lts/pull/3413/files clicking the green button on the right?

Without enough votes I can't merge it and release it.

In this case, having a broken USPS module is a disservice we should fix, we have the fix and we get stuck forever on such a minor thing.

**Review by johnvoncolln — APPROVED** [review]
Looks good to me!

**Review by ADDISON74 — APPROVED** [review]

**Review by kiatng — APPROVED** [review]
Agreed with @fballiano to merge as First-Class will be deprecated in September.

**Pull request merged**
